process/resultsParser/resultsParser.go

```diff
@@ -0,0 +1,401 @@
+	outcome := parseOutcomeOnSimpleMoveBalance(tx)
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on simple move balance", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome = parseOutcomeOnInvalidTransaction(tx)
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on invalid transaction", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err := parseOutcomeOnEasilyFoundResultWithReturnData(tx.SmartContractResults)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on easily found result with return data: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on easily found result with return data", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeOnSignalError(tx.Logs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on signal error: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on signal error", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeOnTooMuchGasWarning(tx.Logs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on too much gas warning: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on too much gas warning", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeOnWriteLogWhereFirstTopicEqualsAddress(tx.Logs, tx.Sender)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on write log where first topic equals address: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on write log where first topic equals address", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeWithFallbackHeuristics(tx, tx.Receivers[0])
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on fallback heuristics", tx.Hash)
+		return outcome, nil
+	}
+	return nil, nil
+}
+
+func parseOutcomeOnSimpleMoveBalance(tx *transaction.ApiTransactionResult) *ResultOutcome {
+	noResults := len(tx.SmartContractResults) == 0
+	var noLogs bool
+	if tx.Logs != nil {
+		noLogs = len(tx.Logs.Events) == 0
+	} else {
+		noLogs = true
+	}
+
+	if noResults && noLogs {
+		return &ResultOutcome{
+			ReturnCode:    None,
+			ReturnMessage: None.String(),
+		}
+	}
+
+	return nil
+}
+
+func parseOutcomeOnInvalidTransaction(tx *transaction.ApiTransactionResult) *ResultOutcome {
+	if tx.Status == transaction.TxStatusInvalid {
+		if tx.Receipt != nil && tx.Receipt.Data != "" {
+			return &ResultOutcome{
+				ReturnCode:    OutOfFunds,
+				ReturnMessage: tx.Receipt.Data,
+			}
+		}
+	}
+
+	// If there's no receipt message, let other heuristics to handle the outcome (most probably, a log with "signalError" is emitted).
+
+	return nil
+}
+
+func parseOutcomeOnEasilyFoundResultWithReturnData(scResults []*transaction.ApiSmartContractResult) (*ResultOutcome, error) {
+	var scr *transaction.ApiSmartContractResult
+	for _, scResult := range scResults {
+		if scResult.Nonce != 0 && scResult.Data != "" && scResult.Data[0] == '@' {
```

Comment:
can you extract these conditions in a variable

Reply:
why do you search for a scr with the nonce greater than 0 ?

Reply:
This is what we knew, since the JS implementation:

https://github.com/multiversx/mx-sdk-js-core/blob/main/src/smartcontracts/resultsParser.ts#L241

Should it be done in a different way?